### PR TITLE
[#526] can now opt out of strong primes to speed up SSL & PAM tests

### PR DESCRIPTION
**Pull request by d-w-moore**

### CODE REVIEW

File: irods/test/setupssl.py [hunk 1]
```diff
@@ -57,14 +65,17 @@ def test(options, args=()):
...
+        dh_strong_primes = '-q' not in options
+        wait_warning = (' This may take a while.' if dh_strong_primes else '')
+        print("Generating new '{}'.{}".format(IRODS_SSL_DIR, wait_warning), file = sys.stderr)
+        ssl_dir_files = create_ssl_dir(use_strong_primes_for_dh_generation = dh_strong_primes)
+        print('ssl_dir_files=', ssl_dir_files, file = sys.stderr)
     
 if __name__ == '__main__':
     import getopt
-    opt, arg_list = getopt.getopt(sys.argv[1:],'x:fh:k')
+    opt, arg_list = getopt.getopt(sys.argv[1:],'x:fh:kq')
```
Comment (korydraughn) [inline irods/test/setupssl.py:76]: Are the options documented somewhere? If not, please add words describing what the options do.

Comment (alanking) [reply irods/test/setupssl.py:76]: `q` is for "quick"? :p

**Review by alanking — COMMENTED** [review]
This is a good find. I think we should consider adding something like this to the irods/irods tests, too.

File: irods/test/setupssl.py (hunk 1 above)
Comment (d-w-moore) [reply irods/test/setupssl.py:76]: yes <sheepish grin>

Comment (d-w-moore) [reply irods/test/setupssl.py]: Done

File: irods/test/setupssl.py [hunk 2]
```diff
@@ -73,7 +73,20 @@ def test(options, args=()):
     
 if __name__ == '__main__':
     import getopt
-    opt, arg_list = getopt.getopt(sys.argv[1:],'x:fh:kq')
+    try:
+        opt, arg_list = getopt.getopt(sys.argv[1:],'x:fh:kq')
+    except getopt.GetoptError:
+        print("""Usage: {sys.argv[0]} [-f] [-h <hostname>] [-k] [-q] [-x <extension>] 
+        -f  Force replacement of the existing SSL directory (/etc/irods/ssl) with a new one, containing newly generated files.
+        -h  In the generated certificate, use the given hostname rather than the value returned from socket.gethostname()
```
Comment (korydraughn) [inline irods/test/setupssl.py:81]: Consider reserving `-h` for printing the help text. Or add a few lines to the README explaining how to view the help text.

**Review by korydraughn — COMMENTED** [review]
Looks good overall.

File: irods/test/setupssl.py (hunk 2 above)
Comment (korydraughn) [reply irods/test/setupssl.py:81]: There's a README in the test directory too in. Either one would be fine IMO.

Comment (alanking) [reply irods/test/setupssl.py:81]: Yeah, let's not change the option now. I think some explanatory text is fine.

I would argue that we should add it to the README - possibly in a separate PR - for expediency.

Comment (alanking) [reply irods/test/setupssl.py:81]: I'm proposing that what you have now is fine and we can add to the README (and the help text itself, *later*) instructions for viewing help text. We can also do it now - whatever works.

Comment (korydraughn) [reply irods/test/setupssl.py:81]: Leave everything as is for and open an issue to add additional information about using the script. The issue can reference this entire conversation.

Comment (d-w-moore) [reply irods/test/setupssl.py:81]: > Leave everything as is for now and open an issue to add additional information about using the script. The issue can reference this entire conversation.

Done

**Comment by d-w-moore** [comment]
I can  either "pound" at will, here, for immediate merging... or we could wait until the merge fest begins later today. But in any case, this will be the next PR to go in.

**Comment by korydraughn** [comment]
If this is ready, squash it. If this doesn't depend on anything, go ahead and pound it.

**Comment by d-w-moore** [comment]
> If this is ready, squash it. If this doesn't depend on anything, go ahead and pound it.

Affirmative, done....

**Pull request merged**